dag.go

```
@@ -733,3 +775,176 @@ func setContextVariablesFromList(c *Config, r types.Resource, values []string, c
+	
+	// Serialize provider data for events
+	providerJSON, jsonErr := json.Marshal(provider)
+	var data []byte
+	if jsonErr == nil {
+		data = providerJSON
+	}
+	
+	// Fire start event
+	fireParserEvent(options, "create", resourceType, resourceID, "start", 0, nil, data)
+	start := time.Now()
+	
+	// Skip if already initialized
+	if provider.Initialized {
+		duration := time.Since(start)
+		fireParserEvent(options, "create", resourceType, resourceID, "success", duration, nil, data)
+		return nil
+	}
+	
+	// Validate required fields
+	if provider.Metadata().Name == "" {
+		err := fmt.Errorf("provider name cannot be empty")
+		fireParserEvent(options, "create", resourceType, resourceID, "error", time.Since(start), err, data)
+		return err
+	}
+	if provider.Source == "" {
+		err := fmt.Errorf("provider source cannot be empty")
+		fireParserEvent(options, "create", resourceType, resourceID, "error", time.Since(start), err, data)
+		return err
+	}
+
+	// Find plugin by source
+	plugin, err := registry.findPluginBySource(provider.Source)
+	if err != nil {
+		err = fmt.Errorf("failed to find plugin for source '%s': %w", provider.Source, err)
+		fireParserEvent(options, "create", resourceType, resourceID, "error", time.Since(start), err, data)
+		return err
+	}
+	if plugin == nil {
+		err := fmt.Errorf("plugin for source '%s' not found", provider.Source)
+		fireParserEvent(options, "create", resourceType, resourceID, "error", time.Since(start), err, data)
+		return err
+	}
+
+	// Get config type from plugin
+	configType := plugin.GetConfigType()
+	if configType == nil {
+		err := fmt.Errorf("plugin for source '%s' does not define a configuration type", provider.Source)
+		fireParserEvent(options, "create", resourceType, resourceID, "error", time.Since(start), err, data)
+		return err
+	}
+
+	// Convert hcl.Body config to concrete type if config is provided
+	if provider.Config != nil {
+		if configBody, ok := provider.Config.(hcl.Body); ok {
```

This cast looks like it silently fails, we should panic if this does not succede as it should not be possible that this can fail